fat/fat2/transactionbatch.go

```diff
@@ -0,0 +1,141 @@
+package fat2
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Factom-Asset-Tokens/factom"
+	"github.com/Factom-Asset-Tokens/fatd/fat"
+)
+
+// Transaction represents a fat2 transaction, which can be a list of one or more
+// sub-transactions to be executed in order
+type TransactionBatch struct {
+	Version      uint          `json:"version"`
+	Transactions []Transaction `json:"transactions"`
+	fat.Entry
+}
+
+// NewTransactionBatch returns a TransactionBatch initialized with the given entry.
+func NewTransactionBatch(entry factom.Entry) TransactionBatch {
+	return TransactionBatch{Entry: fat.Entry{Entry: entry}}
+}
+
+type transactionBatch TransactionBatch
+
+// MarshalJSON marshals the TransactionBatch content field as JSON, but will raise an error if
+// the batch fails the checks in ValidData()
+func (t TransactionBatch) MarshalJSON() ([]byte, error) {
+	if err := t.ValidData(); err != nil {
+		return nil, err
+	}
+	return json.Marshal(transactionBatch(t))
+}
+
+func (t TransactionBatch) String() string {
+	data, err := t.MarshalJSON()
+	if err != nil {
+		return err.Error()
```

Comment:
I do this, and reading it now I hate it. because string shouldn't place an error string where you expect a marshalled json string. but its not a big deal. This would be less of an issue if we didn't have a custom marshaler that checked for data validity. I'm thinking about this out loud...

Reply:
I do agree it would be weird for an error string to be returned from a `String()` call, as there isn't much you can do to detect that something went wrong.

However, I don't feel too too strongly about it. Up to you if you think it should stay here for consistency or not.

Reply:
I believe the correct way to solve this is to write a proper string function that doesn't depend on MarshalJSON and won't error or panic. Adding TODOs in my code.

Reply:
String really should return something human readable, and JSON is only borderline human readable.